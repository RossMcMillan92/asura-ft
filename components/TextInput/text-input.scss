@import '../../styles/settings/pallete.scss';

.text-input {
  width: 100%;
  padding: 10px;
  font-size: 19px;
  color: $base-text-color;
  border: solid 2px $base-ui-color;
  border-radius: 3px;
  transition: border-color 0.25s ease-in-out;

  &:focus {
    border-color: lighten(desaturate($base-theme-color, 10%), 15%);
  }
}
