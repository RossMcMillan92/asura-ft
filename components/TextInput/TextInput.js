import './text-input.scss'

import PropTypes from 'prop-types'
import React from 'react'
import classnames from 'classnames'

const TextInput = ({ className, id, label, name, onChange, ...otherProps }) => (
  <input
    {...otherProps}
    aria-label={label}
    className={classnames('text-input', className)}
    id={id}
    type="text"
    name={name}
    onChange={onChange}
  />
)

TextInput.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func
}

export default TextInput
