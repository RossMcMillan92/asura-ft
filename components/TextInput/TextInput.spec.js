import { mount, shallow } from 'enzyme'

import React from 'react'
import TextInput from './TextInput'
import { act } from 'react-dom/test-utils'

const props = {
  id: 'text-input-id',
  label: 'Text input label',
  name: 'text-input-name',
  onChange: jest.fn()
}
const getWrapper = additionalProps =>
  mount(<TextInput {...props} {...additionalProps} />)
const getShallowWrapper = additionalProps =>
  shallow(<TextInput {...props} {...additionalProps} />)

describe('TextInput', () => {
  it('matches the snapshot', () => {
    const wrapper = getShallowWrapper()
    expect(wrapper).toMatchSnapshot()
  })

  it('adds an aria-label tag to the input', () => {
    const wrapper = getWrapper()
    expect(wrapper.find('input').prop('aria-label')).toBe(props.label)
  })

  describe('when changed', () => {
    it('calls the passed onChange prop', () => {
      const wrapper = getWrapper()
      act(() => {
        wrapper.find('input').simulate('change')
      })
      expect(props.onChange).toBeCalled()
    })
  })

  describe('when a className prop is passed', () => {
    it('appends the class to the className', () => {
      const className = 'additional-styling'
      const wrapper = getWrapper({ className })
      expect(wrapper.find('.text-input').prop('className')).toContain(
        `text-input ${className}`
      )
    })
  })
})
