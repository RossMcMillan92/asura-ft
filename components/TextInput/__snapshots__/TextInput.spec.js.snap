// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput matches the snapshot 1`] = `
<input
  aria-label="Text input label"
  className="text-input"
  id="text-input-id"
  name="text-input-name"
  onChange={[MockFunction]}
  type="text"
/>
`;
